packages/components/src/select-control/index.tsx

```diff
@@ -164,6 +166,14 @@ function UnforwardedSelectControl(
  * };
  * ```
  */
-export const SelectControl = forwardRef( UnforwardedSelectControl );
+export const SelectControl = forwardRef( UnforwardedSelectControl ) as <
+	V extends string,
+>(
+	props: WordPressComponentProps<
+		SelectControlProps< V >,
+		'select',
+		false
+	> & { ref?: React.Ref< HTMLSelectElement > }
+) => ReturnType< typeof UnforwardedSelectControl >;
```

Comment:
I see what you were going for here, but I think I'd simplify the return type to just `React.JSX.Element | null`. That's the only thing that a component can ever return anyway, and it looks clearer to me this way. I don't really mind though, this is a nit.

Reply:
That makes sense, thanks 🙏 [2f195c2](https://github.com/WordPress/gutenberg/pull/64069/commits/2f195c27ecf036e1f7359323fbaf87e53478071c)